### DISCUSSION TITLE
Update spline widget

### DISCUSSION THREAD

**Discussion by HametAl** [pyvista/pyvista#4039] [Q&A]
How can I plot a spline widget and then update it by adding a new point. For example, I plot a spline widget by three points, and then update it and replot it by four points.

**Comment by banesullivan** [comment]
I think you can just update `SetNumberOfHandles` on the widget:

```py
widget = plotter.add_spline_widget(..., n_handles=3)
...
widget.SetNumberOfHandles(4)
```

**Reply by HametAl** [reply]
Thanks for your reply. Don't I need to add a point first and then handle? if I add new point, I have to redraw the spline widget, and the previous widget remains on the plotter. I tried to give the widget a name to be able to update it like a mesh, but I think widgets don't get name
